### PR TITLE
Expose an Elasticsearch client constructor in plugin context

### PR DESCRIPTION
**Pull request by Aschen**
## What does this PR do ?

Plugin developers may want to directly access their Elasticsearch cluster from a plugin.  
Up to now, plugin developers had to instantiate an Elasticsearch client using the [official ES package](https://github.com/elastic/elasticsearch-js).

This PR expose a preconfigured Elasticsearch client connected to the same ES cluster as the one provided in Kuzzle configuration so plugin developer can instantiate a client when they need to send request directly to the ES cluster.

```js
const esClient = new context.constructors.ESClient();

await esClient.info();
```

### Other changes

  - Adds a functional test plugin to test plugin features
  - HTTP verb are now case insensitive

### CODE REVIEW

**Comment by kuzzle** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [elasticsearch.js#L1546](https://github.com/kuzzleio/kuzzle/blob/2a74028d3645095ab4d8bdf1074329a7d7196d53/lib/services/elasticsearch.js#L1546): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [elasticsearch.js#L1587](https://github.com/kuzzleio/kuzzle/blob/2a74028d3645095ab4d8bdf1074329a7d7196d53/lib/services/elasticsearch.js#L1587): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [elasticsearch.js#L1665](https://github.com/kuzzleio/kuzzle/blob/2a74028d3645095ab4d8bdf1074329a7d7196d53/lib/services/elasticsearch.js#L1665): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [elasticsearch.js#L1546](https://github.com/kuzzleio/kuzzle/blob/64760a30bca50fbbe21bbf6831455e68c75d0b17/lib/services/elasticsearch.js#L1546): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [elasticsearch.js#L1587](https://github.com/kuzzleio/kuzzle/blob/64760a30bca50fbbe21bbf6831455e68c75d0b17/lib/services/elasticsearch.js#L1587): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [elasticsearch.js#L1665](https://github.com/kuzzleio/kuzzle/blob/64760a30bca50fbbe21bbf6831455e68c75d0b17/lib/services/elasticsearch.js#L1665): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [elasticsearch.js#L1546](https://github.com/kuzzleio/kuzzle/blob/ef751178d04badad93b99637b44ef6dfa8d85001/lib/services/elasticsearch.js#L1546): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [elasticsearch.js#L1587](https://github.com/kuzzleio/kuzzle/blob/ef751178d04badad93b99637b44ef6dfa8d85001/lib/services/elasticsearch.js#L1587): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [elasticsearch.js#L1665](https://github.com/kuzzleio/kuzzle/blob/ef751178d04badad93b99637b44ef6dfa8d85001/lib/services/elasticsearch.js#L1665): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [elasticsearch.js#L1547](https://github.com/kuzzleio/kuzzle/blob/617834aaac23a0e76562ad00dc3568e3681000b1/lib/services/elasticsearch.js#L1547): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [elasticsearch.js#L1588](https://github.com/kuzzleio/kuzzle/blob/617834aaac23a0e76562ad00dc3568e3681000b1/lib/services/elasticsearch.js#L1588): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [elasticsearch.js#L1666](https://github.com/kuzzleio/kuzzle/blob/617834aaac23a0e76562ad00dc3568e3681000b1/lib/services/elasticsearch.js#L1666): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: doc/2/plugins/plugin-context/constructors/elasticsearch/index.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+code: true
+type: page
+title: Elasticsearch
+---
+
+# Elasticsearch
+
+Instantiates an embedded [Elasticsearch client](https://github.com/elastic/elasticsearch-js) with the same configuration as the one provided in the `.kuzzlerc` file.  
+
+This client can be used to send raw Elasticsearch request.
```
Comment (Yoann-Abbes) [inline doc/2/plugins/plugin-context/constructors/elasticsearch/index.md]:
```suggestion
This client can be used to send raw Elasticsearch requests.  
```

File: doc/2/plugins/plugin-context/constructors/elasticsearch/index.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+title: Elasticsearch
+---
+
+# Elasticsearch
+
+Instantiates an embedded [Elasticsearch client](https://github.com/elastic/elasticsearch-js) with the same configuration as the one provided in the `.kuzzlerc` file.  
+
+This client can be used to send raw Elasticsearch request.  
+
+See [Elasticsearch official documentation](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/api-reference.html) for more informations.
```
Comment (Yoann-Abbes) [inline doc/2/plugins/plugin-context/constructors/elasticsearch/index.md]:
```suggestion
See [Elasticsearch official documentation](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/current/api-reference.html) for more information.
```

File: features-sdk/PluginContext.feature [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  # constructors.ESClient ======================================================
+
+  @mappings
+  Scenario: Instantiate a new embedded ES Client and use it
+    Given an existing collection "nyc-open-data":"yellow-taxi"
+    When I successfully call the route "functional-test-plugin/constructors":"ESClient" with args:
+    | _id | "es-document" |
+    | body | { "from": "embedded-es-client" } |
+    | index | "&nyc-open-data.yellow-taxi" |
+    Then The document "es-document" content match:
```
Comment (Yoann-Abbes) [inline features-sdk/PluginContext.feature]:
```suggestion
    Then The document "es-document" content matches:
```

**Comment by kuzzle** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [elasticsearch.js#L1548](https://github.com/kuzzleio/kuzzle/blob/55c9545f5a84246e3557a7e213891a507564b74e/lib/services/elasticsearch.js#L1548): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [elasticsearch.js#L1589](https://github.com/kuzzleio/kuzzle/blob/55c9545f5a84246e3557a7e213891a507564b74e/lib/services/elasticsearch.js#L1589): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [elasticsearch.js#L1667](https://github.com/kuzzleio/kuzzle/blob/55c9545f5a84246e3557a7e213891a507564b74e/lib/services/elasticsearch.js#L1667): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [elasticsearch.js#L1548](https://github.com/kuzzleio/kuzzle/blob/77e4ba20d90e05376895e6a4807f3f9b58acb18d/lib/services/elasticsearch.js#L1548): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [elasticsearch.js#L1589](https://github.com/kuzzleio/kuzzle/blob/77e4ba20d90e05376895e6a4807f3f9b58acb18d/lib/services/elasticsearch.js#L1589): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [elasticsearch.js#L1667](https://github.com/kuzzleio/kuzzle/blob/77e4ba20d90e05376895e6a4807f3f9b58acb18d/lib/services/elasticsearch.js#L1667): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [elasticsearch.js#L1548](https://github.com/kuzzleio/kuzzle/blob/4c0d3de447ae6c733d24ad77d4e5315970c2417b/lib/services/elasticsearch.js#L1548): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [elasticsearch.js#L1589](https://github.com/kuzzleio/kuzzle/blob/4c0d3de447ae6c733d24ad77d4e5315970c2417b/lib/services/elasticsearch.js#L1589): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [elasticsearch.js#L1667](https://github.com/kuzzleio/kuzzle/blob/4c0d3de447ae6c733d24ad77d4e5315970c2417b/lib/services/elasticsearch.js#L1667): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by kuzzle** [comment]
SonarQube analysis reported 3 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [elasticsearch.js#L1548](https://github.com/kuzzleio/kuzzle/blob/9c79975d8ac1cb26f5551a10913ef491ca8609ad/lib/services/elasticsearch.js#L1548): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [elasticsearch.js#L1589](https://github.com/kuzzleio/kuzzle/blob/9c79975d8ac1cb26f5551a10913ef491ca8609ad/lib/services/elasticsearch.js#L1589): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)
1. ![INFO][INFO] [elasticsearch.js#L1667](https://github.com/kuzzleio/kuzzle/blob/9c79975d8ac1cb26f5551a10913ef491ca8609ad/lib/services/elasticsearch.js#L1667): Complete the task associated to this TODO comment. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Pull request merged**